doc/cephfs/purge-queue.rst

```diff
@@ -0,0 +1,83 @@
+============
+Purge Queue
+============
+
+MDS maintains a data structure known as **Purge Queue** which is responsible
+for managing and executing the sequential deletion of files.
+There is a Purge queue for every MDS rank. Purge queues consist of purge items
+which contain nominal information from the inodes such as size and the layout
+(i.e. all other un-needed metadata information is discarded making it
+independent of all metadata structures).
+
+Deletion process
+================
+
+When a client requests deletion of a directory (say ``rm -rf``):
+
+- MDS queues the files and subdirectories (purge items) from journal in the
+  purge queue.
+- Processes the deletion of inodes in background in small and manageable
+  chunks.
+- MDS instructs underlying OSDs to clean up the associated objects in data
+  pool.
+- Updates the journal.
+
+.. note:: If the users delete the files more quickly than the
+          purge queue can process then the data pool usage might increase
+          substantially over time. In extreme scenarios, the purge queue
+          backlog can become so huge that it can slacken the capacity reclaim
+          and the linux ``du`` command for CephFS might report inconsistent
+          data compared to the CephFS Data pool.
+
+There are a few tunable configs that MDS uses internally to throttle purge
+queue processing:
+
+.. confval:: filer_max_purge_ops 
+.. confval:: mds_max_purge_files
+.. confval:: mds_max_purge_ops
+.. confval:: mds_max_purge_ops_per_pg
+
+.. note:: Generally, the defaults are adequate for most clusters. However, in
+          case of huge clusters, if the need arises, values might be tuned to
+          4-5 times of the default value as a starting point and further 
```

Comment:
4x-5x for all the above configurations?

Reply:
You should also provide a sample configuration.

Reply:
> 4x-5x for all the above configurations?

So, the users can start with the most basic one i.e. setting `filer_max_purge_ops` to 40-50 which mostly should work. If it doesn't then rest three mentioned above can be tuned to 4x-5x.

Reply:
> You should also provide a sample configuration.

I can but it's usually just one or two conf val changes (as I mentioned above). Do you want me to add example of how it's done in the doc?

Reply:
Yeh, I think so. Providing a sample would help users not to second guess. The configs may not be perfect for their cluster, but it provides an understanding of what all to change.

Reply:
> Yeh, I think so. Providing a sample would help users not to second guess. The configs may not be perfect for their cluster, but it provides an understanding of what all to change.

Added some examples, PTAL.